core/src/main/java/nu/yona/server/subscriptions/service/UserUpdateService.java

```diff
@@ -301,39 +333,39 @@ private EncryptedUserData retrieveUserEncryptedDataInNewCryptoSession(User origi
 	}
 
 	@Transactional
-	public void addBuddy(UserDto user, BuddyDto buddy)
+	public void addBuddy(User userEntity, BuddyDto buddy)
 	{
-		Require.that(user != null && user.getId() != null, InvalidDataException::emptyUserId);
+		Require.that(userEntity != null && userEntity.getId() != null, InvalidDataException::emptyUserId);
 		Require.that(buddy != null && buddy.getId() != null, InvalidDataException::emptyBuddyId);
+		userAssertionService.assertUserEntityLockedForUpdate(userEntity);
 
-		updateUser(user.getId(), userEntity -> {
-			userAssertionService.assertValidatedUser(userEntity);
+		userAssertionService.assertValidatedUser(userEntity);
 
-			Buddy buddyEntity = buddyRepository.findById(buddy.getId())
-					.orElseThrow(() -> InvalidDataException.missingEntity(Buddy.class, buddy.getId()));
-			userEntity.addBuddy(buddyEntity);
+		Buddy buddyEntity = buddyRepository.findById(buddy.getId())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "buddy" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS2259)

Reply:
Cannot be null. This is asserted on line 339.